### ISSUE TITLE
NetworkController extends from Application

### ISSUE THREAD

**Issue by MaartenS** [Adyen/AdyenCheckout-android#3]
The NetworkController from the SDK inherits from Application. In this way it's not really possible to include the SDK as a separate android module in your project. Most Android apps have already  an Application class and as you might know, you're only allowed to have one Application in your project. Otherwise a manifest merge will result in a conflict. (http://developer.android.com/tools/building/manifest-merge.html)

In the current state the SDK is not really useful. I still need to implement my own network controller to make it work.

**Comment by FarhanAhmad** [comment]
Or you can extend your application class from NetworkController. In this way, your class will also behave like application class.

**Comment by fgatti675** [comment]
Yes, terrible design decision. I just don't understand how this can be a production SDK

**Comment by chrischen-melb** [comment]
terrible design decision +1
